ground-ingest/src/main/java/edu/berkeley/ground/ingest/TestNoopPublisher.java

```diff
@@ -0,0 +1,79 @@
+package edu.berkeley.ground.ingest;
+
+import gobblin.publisher.*;
+
+import java.io.IOException;
+import java.util.Collection;
+
+import lombok.extern.slf4j.Slf4j;
+
+import gobblin.configuration.State;
+import gobblin.configuration.WorkUnitState;
+
+
+/**
+ * An no-op optimistic publisher that just inspects the state of all workunits
+ * that are successful and marks them committed.
+ */
+@Slf4j
+public class TestNoopPublisher extends DataPublisher {
```

Comment:
All tests should be in `src/test/...` where the `...` reflects the same directory hierarchy that you had for the source code.